packages/backend/src/kit/models/kitManager.ts

```diff
@@ -0,0 +1,36 @@
+import { Logger } from '@map-colonies/js-logger';
+import { inject, injectable } from 'tsyringe';
+import { KitMetadata } from '@map-colonies/detiler-common';
+import { REDIS_KITS_HASH_PREFIX, SERVICES } from '../../common/constants';
+import { RedisClient } from '../../redis';
+import { KitAlreadyExistsError } from './errors';
+import { Kit } from './kit';
+
+@injectable()
+export class KitManager {
+  public constructor(@inject(SERVICES.LOGGER) private readonly logger: Logger, @inject(SERVICES.REDIS) private readonly redis: RedisClient) {}
+
+  public async getAllKits(): Promise<KitMetadata[]> {
+    this.logger.info('getting all kits');
+
+    const keys = await this.redis.keys(`${REDIS_KITS_HASH_PREFIX}:*`);
+
+    const kits = (await Promise.all(keys.map(async (key) => this.redis.hGetAll(key)))) as KitMetadata[];
```

Comment:
Question: 
If one fetch fail, do you want this promise to totally reject? 
Else, I think you need Promise.allSettled.